Use a smaller set of peerDependencies

brianhelba opened the pull request:
peerDependencies should contain only those packages that this directly depends on, which is just eslint-config-semistandard. Other eslint packages will be transitive dependencies.

This will make this package much more flexible to use and upgrade with other eslint versions in the future.

brianhelba committed:
Use a smaller set of peerDependencies

peerDependencies should contain only those packages that this directly
depends on, which is just eslint-config-semistandard. Other eslint packages
will be transitive dependencies.

This will make this package much more flexible to use and upgrade with
other eslint versions in the future.

brianhelba commented:
@jbeezley PTAL.

I have more PRs coming after this, but this will make the eventual release and upgrade process easier for Girder and other plugins that use this.

jbeezley commented:
peerDependencies are intended to specify dependencies that the package doesn't directly depend on.  In this case, the packages you removed [aren't transitive dependencies](https://github.com/Flet/eslint-config-semistandard/blob/master/package.json) and won't be installed automatically.

More generally, you can ask why eslint plugins and configs are always provided as peerDependencies rather than normal dependencies?  There is some discussion of it in https://github.com/eslint/eslint/issues/2518.

zachmullen commented:
In case anyone else was like me and didn't understand the distinction very well, [this was helpful](https://nodejs.org/en/blog/npm/peer-dependencies/).

jbeezley commented:
That's a nice explanation.  It should be noted that it is out of date though.  npm no longer tries to automatically  resolve and install peerDependencies, instead it is up to the top level package to bring them in.  This is crux of the complaint that @brianhelba and others in the thread I mentioned are raising.

jbeezley commented:
To summarize a meat space conversion, the idea here is to rely on the peer dependencies of our peer dependencies instead of locking down versions here.  This will allow packages using this config to update eslint and plugins more flexibly without having to make changes here.  The peer dependencies will remain and will still have to be installed by users of this config (as described in the README).  In light of this, I'm okay with this change.

brianhelba committed:
Make the peerDependencies specify a minimum version

This allows packages that utilize eslint-config-girder to upgrade the
deeper set of eslint dependencies without first having to release a new
eslint-config-girder.

brianhelba commented:
Indeed, the removed packages get depended on transitively, but would never have been installed automatically. I clarified this with @jbeezley in-person.

See one final change, which relaxes the peerDependency against `eslint-config-semistandard`, to allow it to be upgraded (at their own risk) by users of this package, without having to first publish a new release of this package.

brianhelba commented:
@jbeezley Please publish a patch release of this ASAP, so I can get a Girder branch pushed out and into review.
